packages/components/src/border-box-control/border-box-control-linked-button/component.tsx

```diff
@@ -27,7 +27,7 @@ const BorderBoxControlLinkedButton = (
 		<Tooltip text={ label }>
 			<View className={ className }>
 				<Button
-					{ ...buttonProps }
+					{ ...( buttonProps as Parameters< typeof Button >[ 0 ] ) }
```

Comment:
Sorry to expand the scope outside of `components/src/button/`

This corrects a TS error:

```
ackages/components/src/border-box-control/border-box-control-linked-button/component.tsx:29:6 - error TS2322: Type '{ isSmall: true; icon: Element; iconSize: number; "aria-label": string; ref: ForwardedRef<any>; onClick: () => void; onChange?: FormEventHandler<HTMLDivElement> | undefined; ... 250 more ...; as?: keyof IntrinsicElements | undefined; }' is not assignable to type 'Omit<Pick<DetailedHTMLProps<AnchorHTMLAttributes<HTMLAnchorElement>, HTMLAnchorElement>, "key" | keyof AnchorHTMLAttributes<...>> | Pick<...>, "as" | keyof ButtonProps>'.
  Types of property 'onChange' are incompatible.
    Type 'FormEventHandler<HTMLDivElement> | undefined' is not assignable to type 'FormEventHandler<HTMLAnchorElement> | FormEventHandler<HTMLButtonElement> | undefined'.
      Type 'FormEventHandler<HTMLDivElement>' is not assignable to type 'FormEventHandler<HTMLAnchorElement> | FormEventHandler<HTMLButtonElement> | undefined'.
        Type 'FormEventHandler<HTMLDivElement>' is not assignable to type 'FormEventHandler<HTMLAnchorElement>'.
          Type 'HTMLDivElement' is missing the following properties from type 'HTMLAnchorElement': charset, coords, download, hreflang, and 21 more.

```

Reply:
Great idea, committed in https://github.com/WordPress/gutenberg/pull/46997/commits/31cbdab15c013bae56a08cedbc132439a1eb6b8d